src/main/java/uk/gov/companieshouse/api/strikeoffobjections/file/FileTransferApiClient.java

```diff
@@ -167,15 +174,76 @@ public FileTransferApiClientResponse delete(String requestId, String fileId) {
      * @param httpServletResponse The HttpServletResponse to stream the file to
      * @return FileTransferApiClientResponse containing the http status
      */
-    public FileTransferApiClientResponse download(String fileId, HttpServletResponse httpServletResponse) throws IOException {
-        // TODO OBJ-200 replace this dummy response with the implementation.
-        FileTransferApiClientResponse fileTransferApiClientResponse = new FileTransferApiClientResponse();
-        fileTransferApiClientResponse.setFileId(fileId);
-        fileTransferApiClientResponse.setHttpStatus(HttpStatus.OK);
+    public FileTransferApiClientResponse download(String requestId, String fileId, HttpServletResponse httpServletResponse) {
+        String downloadUri = String.format(DOWNLOAD_URI, fileTransferApiURL, fileId);
+
+        return makeApiCall(
+                requestId,
+                //FileTransferOperation
+                () -> restTemplate.execute(
+                        downloadUri,
+                        HttpMethod.GET,
+                        this::handleRequestCallback,
+                        clientHttpResponse -> getClientHttpResponse(httpServletResponse, clientHttpResponse)
+                        ),
+                clientHttpResponse -> getFileTransferApiClientResponse(requestId, clientHttpResponse)
+        );
+    }
+
+    private boolean isFileClean(String requestId, String fileId) {
+        String getUrl = String.format(FILE_TRANSFER_URI_TEMPLATE, fileTransferApiURL, fileId);
+
+        FileTransferApiClientResponse fileDataResponse = makeApiCall(
```

Comment:
can you put the av checking stuff onto another branch so we don't lose it?

Reply:
Will do once this is done and merged in.